numpy/core/src/multiarray/number.c

```diff
@@ -797,6 +797,11 @@ _array_nonzero(PyArrayObject *mp)
         return res;
     }
     else if (n == 0) {
+        if (DEPRECATE("The truth value of an empty array is ambiguous. "
+                      "Returning False, but in future this will result in an error. "
+                      "To check for emptiness, use `a.size > 0` instead.") < 0) {
```

Comment:
Maybe this should change to match the version the release note uses - it is a little clearer.

Reply:
Should I add `will raise an error in the future` in the first line?

Reply:
I think just change the last line to ``To check an array is not empty, use ``array.size > 0```